### PR TITLE
Cache submitter gas block

### PR DESCRIPTION
**Pull request by dwasse**
**Description**
Adds a `gasBlockCache` to submitter so that we can use a cached value in the event that the `HeaderByNumber` RPC call fails.

**Additional context**
Also fixes a bug where we fail to get the gas block properly, but continue with setting gas price and are at risk of getting nullptr.

### Summary by CodeRabbit

- New Feature: Enhanced the transaction submission process in the Etherego platform by introducing a caching mechanism for gas blocks across different chains. This will improve the speed and efficiency of transactions, particularly for users interacting with multiple chains.
- Refactor: Updated the GitHub workflow for better label management. Transitioned from a third-party action to a built-in GitHub action for adding and removing labels, enhancing the maintainability and reliability of our development process.
- Test: Added new tests to ensure the correct functioning of the gas block caching mechanism, ensuring a more robust and reliable transaction submission process.

### CODE REVIEW

**Comment by trajan0x** [comment]
Trying to fix add-label action, as it seems the andymckay/labeler action has been deprecated for over a year: 

image

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -471,13 +471,23 @@
...
+          script: |
+            github.rest.issues.addLabels({
+              issue_number: ${{ needs.pr_metadata.outputs.issue_number }},
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ["needs-go-generate-${{matrix.package}}"]
+            })
       - name: Remove Label
         if: ${{ contains(fromJson(needs.pr_metadata.outputs.labels), format('needs-go-generate-{0}', matrix.package)) && steps.verify-changed-files.outputs.files_changed != 'true' }}
         uses: andymckay/labeler@3a4296e9dcdf9576b0456050db78cfd34853f260
```
[the remark this answers is not in the document]
Comment (trajan0x) [reply .github/workflows/go.yml]: Consolidated into #1466 in 798ef6c4580d0dbc20c5e5ed6f746debb175ba7d

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -471,13 +471,23 @@ jobs:
...
-          repo-token: ${{ secrets.GITHUB_TOKEN }}
-          issue-number: ${{ needs.pr_metadata.outputs.issue_number }}
-
+          script: |
+            github.rest.issues.addLabels({
+              issue_number: ${{ needs.pr_metadata.outputs.issue_number }},
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              labels: ["needs-go-generate-${{matrix.package}}"]
+            })
```
[the remark this answers is not in the document]
Comment (trajan0x) [reply .github/workflows/go.yml]: Consolidated into #1466 in 798ef6c4580d0dbc20c5e5ed6f746debb175ba7d

**Review by trajan0x — APPROVED** [review]

**Review by trajan0x — APPROVED** [review]
LGTM

File: ethergo/submitter/submitter.go [hunk 3]
```diff
@@ -63,6 +64,8 @@ type txSubmitterImpl struct {
 	// to prevent memory leaks, this has a buffer of 1.
 	// callers adding to this channel should not block.
 	retryNow chan bool
+	// gasBlockCache is used to cache the gas block for a given chain.
+	gasBlockCache map[int]*types.Header
```
Comment (trajan0x) [inline ethergo/submitter/submitter.go:68]: Maps are not concurrency safe for writes in golang

**Review by trajan0x — CHANGES_REQUESTED** [review]

File: ethergo/submitter/submitter.go (hunk 3 above)
Comment (trajan0x) [reply ethergo/submitter/submitter.go]: Fixed in 48d19f083e854021cb28a77800b3f4b535ba2bf0

**Comment by trajan0x** [comment]
are agents flaking here? Or did I introduce a regression?

File: ethergo/submitter/submitter.go (hunk 3 above)
Comment (dwasse) [reply ethergo/submitter/submitter.go]: LGTM

**Pull request merged**